### PR TITLE
Wiki Validation Service

### PR DESCRIPTION
**Pull request by philipep-galdino**
# Wiki Validation Service

_A service that validates checks for the wiki content we get from IPFS data on the Indexer Service_

## How should this be tested?

1. Have a IPFS hash run through the Indexer Service
2. Check out if IPFS has been validated correctly
3. If the service worked fine, it's good to go

## Notes or observations

I have had some hard time with typescript on this one and I might have not chosen the best variables names on the reducer that run the checks functions - if you have any suggestion or way to make the code better or readable, awesomely welcome! 

other than that, I think I've gone through close to the best approach we might have here. I didn't want a lot of conditions over the code, wanted something simple to read and also to add another check or remove another check we might have without having a dirty chained condition code. I actually would like to have a second opinion on that and see if I did ok

## Linked issues

closes #00, closes #001, closes https://github.com/EveripediaNetwork/issues/issues/238

### CODE REVIEW

**Comment by philipep-galdino** [comment]
sorry for the delay btw, its been a hard time between the bed and the desk with the cold raping up my nose and headaches lol. i want a sharp review on this one due to this condition of mine while doing it 😅

**Comment by kesar** [comment]
> sorry for the delay btw, its been a hard time between the bed and the desk with the cold raping up my nose and headaches lol. i want a sharp review on this one due to this condition of mine while doing it 😅

please review this code before asking for a code review because from a brief check on mobile I just saw 3 - 4 bugs

File: src/Indexer/Validator/validator.service.ts [hunk 1]
```diff
@@ -1,12 +1,62 @@
...
+      ) {
+        return true
+      }
+
+      return false
+    }
+
+    const checkTags =  (validatingWiki: ValidWiki) => {
+      if (
+        validatingWiki.images.length >= 0 ||
```
Comment (kesar) [inline src/Indexer/Validator/validator.service.ts:42]:
```suggestion
```

length of array can't be negative afaik

**Comment by kesar** [comment]
@AshakaE can you code review this plz? thanks!

**Comment by AshakaE** [comment]
@kesar with respect to the specs, code checks out correctly. Although running the indexer, there were no valid wikis.

**Comment by kesar** [comment]
> @kesar with respect to the specs, code checks out correctly. Although running the indexer, there were no valid wikis.

how its possible that there were not valid wikis? then something is wrong, isnt? I think our latests wikis should be valid

**Comment by AshakaE** [comment]
@kesar & @philipep-galdino I believe the reason is this 
image
😅

**Comment by kesar** [comment]
so userId validation should lowercase both addresses to compare 👍🏻

**Comment by philipep-galdino** [comment]
> @kesar & @philipep-galdino I believe the reason is this image sweat_smile

that is a good catch, tho. I tested it out and it was pretty ok, but seems like for some of them (which have also uppercase letters on the user ID hash) it needs to be lowered case, yeah. damn, nice one!  

I'll change that rn

**Comment by philipep-galdino** [comment]
Had a little tough time on mocking the data and setting up tests in this specific occasion - it was taking too long, so I just push it like this. I'll add an issue so I can work on that after getting on our priorities! 😄 

Whenever you guys give me the ok, I'll merge this!

File: src/Indexer/Validator/validator.service.ts (hunk 1 above)
Comment (philipep-galdino) [reply src/Indexer/Validator/validator.service.ts]: Checked out, yeah. i added it in the new commits

File: src/Indexer/Validator/validator.service.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ class IPFSValidatorService {
         validatingWiki.categories.length === 1 ? true : false
 
     const checkUser = (validatingWiki: ValidWiki) =>
-      validatingWiki.user.id === hashUserId ? true : false
+      validatingWiki.user.id.toLowerCase() === hashUserId ? true : false
```
Comment (kesar) [inline src/Indexer/Validator/validator.service.ts:19]: do lowercase to hashUserId just in case

Comment (philipep-galdino) [reply src/Indexer/Validator/validator.service.ts]: 👍🏾 doneee!

**Comment by kesar** [comment]
code looks good, but make sure you have run the indexer and you are not getting the error it was before

**Pull request merged**